core/src/main/java/org/springframework/security/authentication/AbstractAuthenticationToken.java

```diff
@@ -222,4 +222,16 @@ public String toString() {
 
 		return sb.toString();
 	}
+
+	protected static Integer extractKeyHash(String key) {
```

Comment:
While it provides re-use, this method does not make sense in `AbstractAuthenticationToken` because it knows nothing of a hash key. Instead, we should move this to a private method within each subclass.